packages/relay-compiler/generic/core/RelayWatchmanClient.js

```diff
@@ -17,6 +17,15 @@ const watchman = require('fb-watchman');
 class RelayWatchmanClient {
   _client: any;
 
+  static createIfAvailable(): Promise<?RelayWatchmanClient> {
+    return new Promise((resolve) => {
+      const client = new RelayWatchmanClient();
+      client.on('error', () => resolve(null));
```

Comment:
@leebyron It seems like `fb-watchman` does not work as documented in the link above. The only way I found to handle the error from watchman not being installed is through the `on('error')` callback. I opened an issue on the watchman repo with a minimal example (https://github.com/facebook/watchman/issues/509). How do you think I should proceed?